adapter/service/outlvlswitch/service.go

```diff
@@ -189,35 +193,33 @@ func (s *service) SetBinaryState(value bool) error {
 }
 
 // StartLevelTransition implements starting of the transition with validations and concurrent safety.
-func (s *service) StartLevelTransition(value string, startLvl int, duration time.Duration) error {
+func (s *service) StartLevelTransition(value string, params LevelTransitionParams) error {
```

Comment:
Before proceeding with validations, please call `supportsLevelTransitions`. If not, then return.